src/app/core/services/signing/signing.service.ts

```diff
@@ -28,17 +29,27 @@ export class SigningService {
 
   constructor(private vaultService: VaultService) { }
 
-  signData(data: string, publicKey: string, vaultPassword: string): Observable<SignatureDataModel> {
+  signData(data: string, signingKeyModel: DidKeyEntryModel, vaultPassword: string): Observable<SignatureDataModel> {
     return defer(async () => {
       try {
+        const signingKeyIdParts = signingKeyModel.id.split('#');
+        const didId = signingKeyIdParts[0];
+        const signingKeyAlias = signingKeyIdParts[1];
```

Comment:
Is there a chance the the id of the `signingKeyModel` is malformed and the above breaks?

Reply:
No, only if there is a mistake in our code.